torch/distributed/utils.py

```diff
@@ -49,26 +49,26 @@ def _unpack_kwargs(flat_args: Tuple[Any, ...], kwarg_keys: Tuple[str, ...]) -> T
     kwargs = dict(zip(kwarg_keys, flat_args[-len(kwarg_keys) :]))
     return args, kwargs
 
-def _recursive_to(inputs, target_gpu, use_side_stream_for_tensor_copies):
+def _recursive_to(inputs, target_device, use_side_stream_for_tensor_copies):
     r"""
-    Recursively moves input to the target_gpu.
+    Recursively moves input to the target_device.
     """
 
     def to_map(obj):
         if isinstance(obj, (torch.Tensor, PackedSequence)):
             device = obj.data.device if isinstance(obj, PackedSequence) else obj.device
-            if device == torch.device("cuda", target_gpu):
+            if device == target_device:
                 return (obj,)
             if not use_side_stream_for_tensor_copies:
-                return (obj.to(target_gpu),)
+                return (obj.to(target_device),)
             else:
                 # Perform CPU -> GPU copies in a background stream. This code is
                 # motivated from similar logic in torch/nn/parallel/_functions.py
-                stream = _get_stream(target_gpu)
+                stream = _get_stream(target_device.index)
```

Comment:
Does this actually work for your use case?

Reply:
> Does this actually work for your use case?
Currently，in my use case, input is already on my backend.  But how do I use stream is on my plane.